File: framework/ui-elements/src/displays/components/SetLightMode.js
import React from "react";
import {withStyles} from 'alexandria-ui-elements/src/util/muiStylesCompat';
import AbstractSetLightMode from "../../../gen/displays/components/AbstractSetLightMode";
import SetLightModeNotifier from "../../../gen/displays/notifiers/SetLightModeNotifier";
import SetLightModeRequester from "../../../gen/displays/requesters/SetLightModeRequester";
import DisplayFactory from 'alexandria-ui-elements/src/displays/DisplayFactory';
import {withSnackbar} from "alexandria-ui-elements/src/util/notistackCompat";
import Theme from 'app-elements/gen/Theme';
import Actionable from "./Actionable";

class SetLightMode extends AbstractSetLightMode {

	constructor(props) {
		super(props);
		const theme = Theme.get();
		this.notifier = new SetLightModeNotifier(this);
		this.requester = new SetLightModeRequester(this);
		this.state = {
            ...this.state,
            visible: this.state.appMode == "dark",
		};
	};

	renderActionable() {
	    return (
	        <React.Fragment>
	            {this.renderCookieConsent()}
	            {super.renderActionable()}
	        </React.Fragment>
	    );
	};

	updateMode = () => {
	    Theme.get().setMode("light");
	    this._saveAppModeInCookies("light");
	    this.setState({appMode: "light"});
	};
}

export default withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SetLightMode));
DisplayFactory.register("SetLightMode", withStyles(Actionable.Styles, { withTheme: true })(withSnackbar(SetLightMode)));

File: framework/ui/src/io/intino/alexandria/ui/displays/components/SetLightMode.java
package io.intino.alexandria.ui.displays.components;

import io.intino.alexandria.core.Box;
import io.intino.alexandria.ui.displays.events.actionable.ExecuteEvent;
import io.intino.alexandria.ui.displays.events.actionable.ExecuteListener;
import io.intino.alexandria.ui.displays.notifiers.SetLightModeNotifier;

public class SetLightMode<DN extends SetLightModeNotifier, B extends Box> extends AbstractSetLightMode<DN, B> {
	private ExecuteListener executeListener;

	public SetLightMode(B box) {
		super(box);
	}

	public SetLightMode<DN, B> onExecute(ExecuteListener listener) {
		this.executeListener = listener;
		return this;
	}

	public void execute() {
		notifier.updateMode();
		soul().displays(SetLightMode.class).forEach(d -> d.visible(false));
		soul().displays(SetDarkMode.class).forEach(d -> d.visible(true));
		executeListener.accept(new ExecuteEvent(this, "light"));
	}

}